study-builder/studies/atcp/registration.conf

```diff
@@ -0,0 +1,765 @@
+                      "translations": [
+                        {
+                          "language": "en",
+                          "text": ${i18n.en.registration_gender_male}
+                        }
+                      ]
+                    }
+                  ]
+                },
+                "detailLabelTemplate": null,
+                "allowDetails": false,
+                "exclusive": false
+              },
+              {
+                "stableId": "F",
+                "optionLabelTemplate": {
+                  "templateType": "TEXT",
+                  "templateText": "$registration_gender_female",
+                  "variables": [
+                    {
+                      "name": "registration_gender_female",
+                      "translations": [
+                        {
+                          "language": "en",
+                          "text": ${i18n.en.registration_gender_female}
+                        }
+                      ]
+                    }
+                  ]
+                },
+                "detailLabelTemplate": null,
+                "allowDetails": false,
+                "exclusive": false
+              }
+            ]
+            "stableId": "REGISTRATION_GENDER"
+          },
+          "blockType": "QUESTION",
+          "shownExpr": null
+        },
+        {
+          "question": {
+            include required("../snippets/date-question-dob-required.conf"),
+            "stableId": "REGISTRATION_DOB",
+            "hideNumber": true,
+            "renderMode": "SINGLE_TEXT",
+            "displayCalendar": true,
+            "promptTemplate": {
+              "templateType": "HTML",
+              "templateText": "$registration_dob",
+              "variables": [
+                {
+                  "name": "registration_dob",
+                  "translations": [
+                    {
+                      "language": "en",
+                      "text": ${i18n.en.registration_dob}
+                    }
+                  ]
+                }
+              ]
+            },
+            "validations": [
+              {
+                "ruleType": "REQUIRED",
+                "hintTemplate": {
+                  "templateType": "TEXT",
+                  "templateText": "$registration_dob_req_hint",
+                  "variables": [
+                    {
+                      "name": "registration_dob_req_hint",
+                      "translations": [
+                        {
+                          "language": "en",
+                          "text": ${i18n.en.registration_dob_req_hint}
+                        }
+                      ]
+                    }
+                  ]
+                }
+              }
+            ]
+          },
+          "blockType": "QUESTION",
+          "shownExpr": null
+        },
+        {
+          "question": {
+            include required( "../snippets/text-question.conf"),
```

Comment:
We have a an address component that has autocomplete and verification against EasyPost, and offers suggestions.  Strongly encourage you to use that. Please do let us know if the address component does not serve your purposes in some way. Here is a place in one of our existing study configuration where we use it:
https://github.com/broadinstitute/ddp-study-server/blob/12ce673b3d6a4eafaa312af3d84e3004aa87517a/study-builder/studies/angio/release.conf#L269